dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.cpp

```diff
@@ -30,8 +30,6 @@
 #include "DYNTrace.h"
 
 using boost::shared_ptr;
-using std::string;
-using std::vector;
 
```

Comment:
why removing this?

Reply:
We don't need these lines. The first one takes about 20 chars and one line. Instead I just add "std::" in the only place where it is needed. Using "using" keyword is not a very good practice. It's delicate (never before an include or within an include), makes the reader confusing... Who remembers what shared_ptr stands for in that file where powsybl may use std::smart_pointers and dynawo boost::smart_ones (?) It makes our code less readable and more difficult to maintain... The keyword was introduced mainly to help migrate code written before 2000 that was not using namespaces. Instead of changing thousands of lines as "cerr << endl" you could just add "using std::cerr..." That was more than 20 years back.

Reply:
The keyword "using" may have sense in very little examples like on 'cppreference.com'. But we should avoid it for bigger projects as Dynawo. In fact in that case, for a brand new file, I asked myself... I have 2 main examples in mind : the legacy-iidm file LineInterface.cpp and a new one BusInterface.cpp. In both cases whoever wrote the code uses sometimes "std::string" and sometimes only "string" assuming there was a "using std::string" line above. I wonder why using both ways (?) Since it is confusing to me, I just choosed one: simpler, faster, less confusing. It does not matter much in fact. OK ?

Reply:
I strongly disagree with you on the usage of the word using, I think it makes the code more readable on the contrary.
If in some files there are both using std::string and std::string the second one is a mistake. Let's keep it this way for this file as you already modified it and as you highlighted there are only 2-3 occurences, but in general we should keep those kind of lines in the cpp files.

Reply:
I was afraid in fact that your fist comment was not a question... As I told, it makes small examples easier to read. But C++ was re-designed around 2000, introducing namespaces and a lot more ambition! As you says, I keep the way it is now. As you recognize it is source of mistakes since either you use it, either not, but it is a mistake to use both "styles". If you use it once, why not for other methods like std::is_nan, std::make_pair and sometimes for powsybl::iidm::Bus but not for powsybl::iidm::Lines... You know that you can still use typedef for the same purpose. Why not simplifying Trace::warning and change it to Warning too ?

Reply:
As soon as I have time, I set an example showing how "dangerous" using can be. I will close that point, but I think we still may discuss on that for eternity.

Reply:
I don't need an example and I don't have time to discuss this further, I'm just saying that when a system class is used widely in a cpp it is far more readable to have some using. In dynawo for using std::string, std::vectors and boost::smart_pointers are use extensively in dynawo, this is not something we want to change and I'm asking you to keep it that way.